go-controller/pkg/ovn/pod_selector_address_set.go

```diff
@@ -335,7 +343,7 @@ func (handlerInfo *PodSelectorAddrSetHandlerInfo) addPods(pods ...*v1.Pod) error
 	}
 
 	podIPFactor := 1
-	if config.IPv4Mode && config.IPv6Mode {
+	if handlerInfo.ipv4Mode && handlerInfo.ipv6Mode {
```

Comment:
this function together with `deletePod` call 
`util.GetPodIPsOfNetwork(pod, &util.DefaultNetInfo{})`
is it intended to use default network here?

Reply:
good question - I think it should be using the network to which the address set corresponds.

 I'm running this PR locally and I think the pod selector addresses set features IPs from the wrong network:
```
_uuid               : a914ea2b-14ce-4e34-a639-bb0a1ff16c33
addresses           : ["192.168.100.1", "192.168.100.2", "192.168.100.3"]
external_ids        : {ip-family=v4, "k8s.ovn.org/id"="tenant-blue-network-controller:Namespace:default:v4", "k8s.ovn.org/name"=default, "k8s.ovn.org/owner-controller"=tenant-blue-network-controller, "k8s.ovn.org/owner-type"=Namespace}
name                : a9836740957047443908

# I think this address set should have an IP address of the secondary network, not from the primary
_uuid               : 7ab6702f-a699-470d-818d-a15aa9904934
addresses           : ["10.244.0.14"]
external_ids        : {ip-family=v4, "k8s.ovn.org/id"="tenant-blue-network-controller:PodSelector:default_LS{ML:{app: allowed-client,},}:v4", "k8s.ovn.org/name"="default_LS{ML:{app: allowed-client,},}", "k8s.ovn.org/owner-controller"=tenant-blue-network-controller, "k8s.ovn.org/owner-type"=PodSelector}
name                : a1493214346180864325
```

Here's the net-attach-def / policy I'm using:
```yaml
---
apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"k8s.cni.cncf.io/v1","kind":"NetworkAttachmentDefinition","metadata":{"annotations":{},"name":"tenant-blue","namespace":"default"},"spec":{"config":"{ \"cniVersion\": \"0.4.0\", \"name\": \"tenant-blue\", \"netAttachDefName\": \"default/tenant-blue\", \"topology\": \"layer2\", \"type\": \"ovn-k8s-cni-overlay\", \"subnets\": \"192.168.100.0/24\", \"logFile\": \"/var/log/ovn-kubernetes/ovn-k8s-cni-overlay.log\", \"logLevel\": \"5\", \"logfile-maxsize\": 100, \"logfile-maxbackups\": 5, \"logfile-maxage\": 5 }"}}
  creationTimestamp: "2023-04-12T10:41:26Z"
  generation: 1
  name: tenant-blue
  namespace: default
  resourceVersion: "5904"
  uid: d1e70f5f-4ab2-4d5d-900a-9988746a5f6b
spec:
  config: '{ "cniVersion": "0.4.0", "name": "tenant-blue", "netAttachDefName": "default/tenant-blue",
    "topology": "layer2", "type": "ovn-k8s-cni-overlay", "subnets": "192.168.100.0/24",
    "logFile": "/var/log/ovn-kubernetes/ovn-k8s-cni-overlay.log", "logLevel": "5",
    "logfile-maxsize": 100, "logfile-maxbackups": 5, "logfile-maxage": 5 }'
---
apiVersion: k8s.cni.cncf.io/v1beta1
kind: MultiNetworkPolicy
metadata:
  annotations:
    k8s.v1.cni.cncf.io/policy-for: tenant-blue
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"k8s.cni.cncf.io/v1beta1","kind":"MultiNetworkPolicy","metadata":{"annotations":{"k8s.v1.cni.cncf.io/policy-for":"tenant-blue"},"name":"allow-ports-same-ns","namespace":"default"},"spec":{"ingress":[{"from":[{"podSelector":{"matchLabels":{"app":"allowed-client"}}}]}],"podSelector":{"matchLabels":{"app":"stuff-doer"}}}}
  creationTimestamp: "2023-04-12T10:41:26Z"
  generation: 1
  name: allow-ports-same-ns
  namespace: default
  resourceVersion: "5905"
  uid: bf691987-53fd-4bc8-bddd-07306231b56f
spec:
  ingress:
  - from:
    - podSelector:
        matchLabels:
          app: allowed-client
  podSelector:
    matchLabels:
      app: stuff-doer
```

View from a pod:
```bash
kubectl exec pod1 -- ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0@if30: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1400 qdisc noqueue state UP group default 
    link/ether 0a:58:0a:f4:00:0e brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.244.0.14/24 brd 10.244.0.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::858:aff:fef4:e/64 scope link 
       valid_lft forever preferred_lft forever
3: net1@if32: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1400 qdisc noqueue state UP group default 
    link/ether 0a:58:c0:a8:64:01 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 192.168.100.1/24 brd 192.168.100.255 scope global net1
       valid_lft forever preferred_lft forever
    inet6 fe80::858:c0ff:fea8:6401/64 scope link 
       valid_lft forever preferred_lft forever

```

Reply:
@cathy-zhou

Reply:
> this function together with deletePod call
> util.GetPodIPsOfNetwork(pod, &util.DefaultNetInfo{})
> is it intended to use default network here?